### ISSUE TITLE
Docker image is required to be pushed when use docker-py>=1.3

### ISSUE THREAD

**Issue by DXist** [ansible/ansible-modules-core#1794]
Issue Type:

Bug Report

Ansible Version:

ansible 1.9.2

Ansible Configuration:
## 

Environment:

Ubuntu Utopic
docker-py==1.3.1

Summary:

After I update docker-py to 1.3.1 ansible unsuccessfully tries to find my image in private registry which results following error:

```
failed: [127.0.0.1] => {"changes": ["{\"status\":\"Pulling repository myregistry:5000/myimage\"}\r\n", "{\"errorDetail\":{\"message\":\"Error: image myimage:latest not found\"},\"error\":\"Error: image myimage:latest not found\"}\r\n"], "failed": true, "status": ""}
msg: Unrecognized status from pull.

FATAL: all hosts have already failed -- aborting
```

Steps To Reproduce:
1. prepare simple Dockerfile
2. docker build -t myregistry:5000/myimage .
3. run container using following task and `connection=local` parameter:

```
    docker:
      detach: False
      stdin_open: True
      tty: True
      image: "myregistry:5000/myimage"
      name: "container"
      command: "{{ lookup('env', 'SHELL')}}"
      state: started
```

Expected Results:

Ansible should use an image built and saved locally.

Actual Results:

Ansible tries to pull an image from `myregistry:5000`

**Comment by indrekots** [comment]
I seem to be having the same issue.

**Ansible Version:**
ansible 1.9.2

**Environment:**
Ubuntu 14.04.02
docker-py 1.3.1

```
- name: start docker container
    docker:
      image=my_image
      detach=yes
```

This results in the following error:

```
failed: [192.168.1.129] => {"changes": ["{\"status\":\"Pulling repository my_image\"}\r\n", "{\"errorDetail\":{\"message\":\"Error: image library/my_image:latest not found\"},\"error\":\"Error: image library/my_image:latest not found\"}\r\n"], "failed": true, "status": ""}
msg: Unrecognized status from pull.

FATAL: all hosts have already failed -- aborting
```

Running _docker images_ i can clearly see that _my_image_ with the tag _latest_ is available.

**Comment by kyriesent** [comment]
+1.

It's not just docker-py > 1.3. I'm running 1.1.0.

I can confirm that this same playbook is was working on previous servers, but this is a newly minted server and it's having issues.

**Ansible Version**
Ansible 1.9.2

**Remote Host Environment:**
CoreOS 723.3.0
Python 2.7.8 (PyPy 2.4.0)
docker-py 1.1.0

Sample Tasks

```
- name: Build docker images
  docker_image:
    path: "/opt/{{ item }}"
    name: "seelio/{{ item }}"
    state: build
  with_items:
    - fluentd

- name: Run fluentd docker container
  docker:
    image: seelio/fluentd
    name: fluentd
    links: elasticsearch:elasticsearch
    volumes:
      - "/var/run/docker.sock:/tmp/docker.sock"
      - "/var/lib/docker:/var/lib/docker"
      - "/mnt/vol/fluent:/var/log/fluent"
      - "/opt/fluentd/fluentd.tmpl:/app/config/fluentd.tmpl"
    ports:
      - "42185:42185/udp"
    state: running
    restart_policy: always
```

And the same error:

```
failed: [ec2-54-153-148-140.ap-southeast-2.compute.amazonaws.com] => {"changes": ["{\"status\":\"Pulling repository seelio/fluentd\"}\r\n", "{\"errorDetail\":{\"message\":\"Error: image seelio/fluentd:latest not found\"},\"error\":\"Error: image seelio/fluentd:latest not found\"}\r\n"], "failed": true, "status": ""}
msg: Unrecognized status from pull.
```

I can also confirm that the image seelio/fluentd:latest exists.

**Comment by neocortical** [comment]
Also experiencing this issue. Anyone have a workaround? Downgrading to a previous Ansible release does not seem to help.

**Comment by neocortical** [comment]
Update: Adding the following parameter to my docker command was a successful workaround for me: 

``` go
docker_api_version=1.18
```

See comments here: https://github.com/ansible/ansible-modules-core/issues/1707

EDIT: Possibly not actually a workaround. Worked on one machine, not on another. Something odd is going on in my env. YMMV.

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by chouseknecht** [comment]
wontfix

docker.py is deprecated and replace by docker_container, docker_service and a refactored docker_image module.
